tests/suite/reconfig_test.go

```diff
@@ -0,0 +1,614 @@
+		queryRangeStep        = 5 * time.Second
+		promInstance          framework.PrometheusInstance
+		promPortForwardStopCh = make(chan struct{})
+
+		outFile *os.File
+	)
+
+	BeforeAll(func() {
+		resultsDir, err := framework.CreateResultsDir("reconfig", version)
+		Expect(err).ToNot(HaveOccurred())
+
+		filename := filepath.Join(resultsDir, framework.CreateResultsFilename("md", version, *plusEnabled))
+		outFile, err = framework.CreateResultsFile(filename)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(framework.WriteSystemInfoToFile(outFile, clusterInfo, *plusEnabled)).To(Succeed())
+
+		promCfg := framework.PrometheusConfig{
+			ScrapeInterval: scrapeInterval,
+		}
+
+		promInstance, err = framework.InstallPrometheus(resourceManager, promCfg)
+		Expect(err).ToNot(HaveOccurred())
+
+		k8sConfig := ctlr.GetConfigOrDie()
+
+		if !clusterInfo.IsGKE {
+			Expect(promInstance.PortForward(k8sConfig, promPortForwardStopCh)).To(Succeed())
+		}
+	})
+
+	BeforeEach(func() {
+		teardown(releaseName)
+	})
+
+	AfterAll(func() {
+		teardown(releaseName)
+		close(promPortForwardStopCh)
+		Expect(framework.UninstallPrometheus(resourceManager)).To(Succeed())
+	})
+
+	createUniqueResources := func(resourceCount int, fileName string) error {
+		for i := 1; i <= resourceCount; i++ {
+			nsName := "namespace" + strconv.Itoa(i)
+			// Command to run sed and capture its output
+			//nolint:gosec
+			sedCmd := exec.Command("sed",
+				"-e",
+				"s/coffee/coffee"+nsName+"/g",
+				"-e",
+				"s/tea/tea"+nsName+"/g",
+				fileName,
+			)
+			// Command to apply using kubectl
+			kubectlCmd := exec.Command("kubectl", "apply", "-n", nsName, "-f", "-")
+
+			sedOutput, err := sedCmd.Output()
+			if err != nil {
+				fmt.Println(err.Error() + ": " + string(sedOutput))
+				return err
+			}
+			kubectlCmd.Stdin = bytes.NewReader(sedOutput)
+
+			output, err := kubectlCmd.CombinedOutput()
+			if err != nil {
+				fmt.Println(err.Error() + ": " + string(output))
+				return err
+			}
+		}
+		return nil
+	}
+
+	createResourcesGWLast := func(resourceCount int) error {
+		ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetTimeout)
+		defer cancel()
+
+		for i := 1; i <= resourceCount; i++ {
+			ns := core.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "namespace" + strconv.Itoa(i),
+				},
+			}
+			Expect(k8sClient.Create(ctx, &ns)).To(Succeed())
+		}
+
+		ns := core.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "reconfig",
+			},
+		}
+		Expect(resourceManager.Apply([]client.Object{&ns})).To(Succeed())
+		Expect(resourceManager.ApplyFromFiles(
+			[]string{
+				"reconfig/certificate-ns-and-cafe-secret.yaml",
+				"reconfig/reference-grant.yaml",
+			},
+			ns.Name)).To(Succeed())
+
+		Expect(createUniqueResources(resourceCount, "manifests/reconfig/cafe.yaml")).To(Succeed())
+
+		Expect(createUniqueResources(resourceCount, "manifests/reconfig/cafe-routes.yaml")).To(Succeed())
+
+		time.Sleep(60 * time.Second)
```

Comment:
I intentionally tried to keep this code as close as possible to the original script which included the sleep for 60 seconds. 

Do you think that this would be a situation where a deviation from the original test script would be good?

Reply:
Yes, blind sleeps should always be avoided if possible. The intention of the script is just to wait until pods are ready, which is the goal we want to achieve, however we might do that. Technically the script shouldn't be doing it either.